scripts/fetchArticles.js

```diff
@@ -2,75 +2,53 @@ const request = require('request-promise')
 const retext = require('retext')
 const retextkeywords = require('retext-keywords')
 const nlcstToString = require('nlcst-to-string')
-const Article = require('../models/ArticleModel.js')
+const Article = require('../models/ArticleModel')
 
 // 3 600 000 is 1 hour
 const REFRESH = 3600000
 
-module.exports.startFetchCycle = () => {
-    setInterval(() => {
-        updateNews()
-        console.log('ran') 
-    }, REFRESH)
-}
-
-function getWords(article) {
-    let words = []
-    retext().use(retextkeywords).process(
-        `${article.title} ${article.description}`, (err, file) => {
-            file.data.keywords.forEach((keyword) => {
-                // sanitize it
-                let word = sanitize(nlcstToString(keyword.matches[0].node))
-                words.push(nlcstToString(keyword.matches[0].node))
-            })
-        })
-    //console.log(article)
-    return words
+function sanitize(string) {
+  const newString = string.toLowerCase()
+  // Removes all punctuation
+  return newString.replace(/[^0-9a-z ]/g, '')
 }
 
 function getPhrases(article) {
-    let words = []
-    retext().use(retextkeywords).process(
-        `${article.title} ${article.description}`, (err, file) => {
-            file.data.keyphrases.forEach((phrase) => {
-                words.push(sanitize(phrase.matches[0].nodes.map(nlcstToString).join('')))
-            })
-        })
-    //console.log(article)
-    return words
+  const words = []
+  retext()
+    .use(retextkeywords)
+    .process(`${article.title} ${article.description}`, (err, file) => {
+      file.data.keyphrases.forEach((phrase) => {
+        const newPhrase = phrase.matches[0].nodes
+          .map(nlcstToString)
+          .join('')
+        words.push(sanitize(newPhrase))
+      })
+    })
+  return words
 }
 
-function sanitize(string) {
-    // formats the string to lowercase, and removes all punctuation
-    string = string.toLowerCase()
-    return string.replace(/[^0-9a-z ]/g, '')
+function updateNews() {
+  request(process.env.NEWS_URI)
+    .then((res) => {
+      const headlines = JSON.parse(res).articles
+      headlines.forEach((old) => {
+        Article.findOne({ url: old.url })
+          .then((article) => {
+            if (article) return
+            const newArticle = new Article(old)
+            newArticle.keywords = getPhrases(article)
+            newArticle.votes = 0
+            newArticle.save()
+          })
+          .catch(err => err)
```

Comment:
@VikramBombhi A long time man, but I think the solution is good, but might require mocking a db